app/lib/bing_api_v7.rb

```diff
@@ -0,0 +1,38 @@
+require 'net/https'
+require 'uri'
+require 'json'
+
+class BingApiV7
+  URI::PATH = "https://api.cognitive.microsoft.com/bing/v7.0/search".freeze
+
+  def initialize
+    @api_key = Rails.application.secrets.bing_api_v7
+  end
+
+  def search(company_domain)
+    uri = URI(URI::PATH + "?q=" + CGI.escape(company_domain))
+    request = Net::HTTP::Get.new(uri)
+    request['Ocp-Apim-Subscription-Key'] = @api_key
+
+    response = Net::HTTP.start(uri.host, uri.port, use_ssl: uri.scheme == 'https') do |http|
+      http.request(request)
+    end
+
+    response.each_header do |key, value|
+      puts key + ": " + value if key.start_with?("bingapis-", "x-msedge-")
+    end
+
+    JSON(response.body)
+  end
+
+  def bing_pages_to_model(company)
+    pages = search(company.domain)["webPages"]["value"]
+    
+    pages.each do |page|
+      company.pages.create(page_type: Page::BING_TYPE, title: page["name"],
+                           source_url: page["displayUrl"], status: Page::PENDING_STATUS)
+    end
+    rescue NoMethodError => e  
```

Comment:
don't forget to remove this before merge

Reply:
why?

Reply:
because you are not handling this exception
for example, in case you have code like "pages.cccreate" and this rescue will catch this error, BUT this is bug in code.

Remember: if you are using rescue in most cases it's wrong and you have issue in code. This is a rule.

Reply:
here is no need to catch all `NoMethodError`s
it's just about response data structure and validity, so you need to validate and check it instead of rescueing